rllib/algorithms/cql/torch/cql_torch_rl_module.py

```diff
@@ -0,0 +1,205 @@
+import tree
+from typing import Any, Dict, Optional
+
+from ray.rllib.algorithms.sac.sac_learner import (
+    QF_PREDS,
+    QF_TWIN_PREDS,
+)
+from ray.rllib.algorithms.sac.torch.sac_torch_rl_module import SACTorchRLModule
+
+from ray.rllib.core.columns import Columns
+from ray.rllib.core.models.base import ENCODER_OUT
+from ray.rllib.utils.annotations import override
+from ray.rllib.utils.framework import try_import_torch
+from ray.rllib.utils.typing import TensorType
+
+torch, nn = try_import_torch()
+
+
+class CQLTorchRLModule(SACTorchRLModule):
+    @override(SACTorchRLModule)
+    def _forward_train(self, batch: Dict) -> Dict[str, Any]:
+        # Call the super method.
+        fwd_out = super()._forward_train(batch)
+
+        # Make sure we perform a "straight-through gradient" pass here,
+        # ignoring the gradients of the q-net, however, still recording
+        # the gradients of the policy net (which was used to rsample the actions used
+        # here). This is different from doing `.detach()` or `with torch.no_grads()`,
+        # as these two methds would fully block all gradient recordings, including
+        # the needed policy ones.
+        all_params = list(self.pi_encoder.parameters()) + list(self.pi.parameters())
```

Comment:
Wait, why `pi`? We need to block the q-net gradients (same as in SAC), correct?

Reply:
```
        all_params = (
            list(self.qf.parameters())
            + list(self.qf_encoder.parameters())
            + list(self.qf_twin.parameters())
            + list(self.qf_twin_encoder.parameters())
        )
        for param in all_params:
            param.requires_grad = False
        output["q_curr"] = self.compute_q_values(q_batch_curr)
        for param in all_params:
            param.requires_grad = True
```

Reply:
^ from SAC

Reply:
Well, I don't think so. That was my first impression of it. But using this kind of backward pass let's the CQL loss rise without limit. Which makes sense: The additional loss term is now merely an added constant which is different from time to time, but does not change in regard to its own amount. Removing in this forward pass the `gradient=False` does make the CQL loss term decrease by time - as it should. 

Note, the SAC loss is computed earlier and uses in the `super()._forward_train` exactly the logic you posted above.